post-processor/vagrant/libvirt_test.go

```diff
@@ -70,3 +74,76 @@ func Test_sizeInMegabytes_WithExabytesUnit(t *testing.T) {
 	assertSizeInMegabytes(t, "1234E", 1234*1024*1024*1024*1024)
 	assertSizeInMegabytes(t, "1E", 1*1024*1024*1024*1024)
 }
+
+func Test_ManyFilesInArtifact(t *testing.T) {
+	p := new(LibVirtProvider)
+	ui := testUi()
+	type testCases struct {
+		Files         []string
+		Format        string
+		FilesExpected []string
+	}
+	testcases := []testCases{
+		{
+			[]string{},
+			"qcow2",
+			[]string{},
+		},
+		{
+			[]string{"test"},
+			"vmdk",
+			[]string{"box_0.img"},
+		},
+		{
+			[]string{"test", "/test-1", "test-2"},
+			"qcow2",
+			[]string{"box_0.img", "box_1.img", "box_2.img"},
+		},
```

Comment:
I would suggest adding one extra test with a file that doesn't match the image's name to ensure it doesn't end-up in the `FilesExpected` at the end.

Reply:
well seen. The new test detected a bug. I separate filter and the copy of artifact for keep continue sequence, if we have bad file.

Reply:
Looking at the code, I'd like more context on the bug you solve? It seems to me that the two loops can effectively be condensed into one, I'm not sure I understand why we need to do this in two passes

Reply:
The solved bug is :
Files: []string{"test", "efivars.fd", "test-1", "test-2"} given []string{"box_0.img", "box_2.img", "box_3.img"} and not []string{"box_0.img", "box_1.img", "box_2.img"}.
Now, it's corrected.

Reply:
I merge the two loops, too.

Reply:
Oh, is that because in a previous version you were breaking out of the loop as soon as you encountered a file that did not match the pattern?

Reply:
Yes, that's it.